pkg/chartmuseum/server/multitenant/cache.go

```diff
@@ -357,6 +357,11 @@ func (server *MultiTenantServer) initCacheEntry(log cm_logger.LoggingFn, repo st
 
 	if server.ExternalCacheStore == nil {
 		var ok bool
+		if server.KeepChartAlwaysUpToDate {
+			// waiting for the index regeneration and keep the entry index always up-to-date
+			server.Tenants[repo].RegenerationLock.Lock()
+			defer server.Tenants[repo].RegenerationLock.Unlock()
+		}
```

Comment:
To rollback the old logic we may not need to add the new option `KeepChartAlwaysUpToDate ` , we only need to check whether the `cache-interval` is set or not , and we will use the old logic if `cache-interval` not set , otherwise , set the provided cache interval, and enables the event emitter.

My original purpose adding the new option `KeepChartAlwaysUpToDate ` is that if we can have some mechanisms to always get the latest chart with the event emitter is enabled.